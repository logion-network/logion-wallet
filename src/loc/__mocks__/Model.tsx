import {
    rejectProtectionRequest,
    acceptProtectionRequest,
} from './ModelMock';

export {
    rejectProtectionRequest,
    acceptProtectionRequest,
};
