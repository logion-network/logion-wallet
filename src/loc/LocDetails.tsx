import { useParams } from "react-router";
import { UUID } from "@logion/node-api/dist/UUID";
import { LocType } from "@logion/node-api/dist/Types";

import ContextualizedLocDetails from "./ContextualizedLocDetails";
import { UserLocContextProvider } from "./UserLocContext";
import UserContextualizedLocDetails from "./UserContextualizedLocDetails";
import { LegalOfficerLocContextProvider } from "./LegalOfficerLocContext";
import { Viewer } from "src/common/CommonContext";

export interface Props {
    backPath: string;
    detailsPath: (locId: UUID, type: LocType) => string;
    viewer: Viewer;
}

export default function LocDetails(props: Props) {
    const locId: UUID = new UUID(useParams<"locId">().locId);

    if (props.viewer === "LegalOfficer") {
        return (
            <LegalOfficerLocContextProvider
                locId={ locId }
                backPath={ props.backPath }
                detailsPath={ props.detailsPath }
            >
                <ContextualizedLocDetails/>
            </LegalOfficerLocContextProvider>
        )
    } else {
        return (
            <UserLocContextProvider
                locId={ locId }
                backPath={ props.backPath }
                detailsPath={ props.detailsPath }
            >
                <UserContextualizedLocDetails/>
            </UserLocContextProvider>
        )
    }
}
