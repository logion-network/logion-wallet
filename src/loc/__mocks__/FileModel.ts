export { getAllDeliveries, getAllCollectionDeliveries } from './FileModelMock';
