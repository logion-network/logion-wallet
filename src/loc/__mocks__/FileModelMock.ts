export let getAllDeliveries = jest.fn().mockResolvedValue({});
