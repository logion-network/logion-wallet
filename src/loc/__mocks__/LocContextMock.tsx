import { LocData, CollectionItem } from "@logion/client";
import { UUID, ValidAccountId } from "@logion/node-api";

import { LocItem } from "../LocItem";

let locState: any = {
    isLogionIdentity: () => false,
    isLogionData: () => true,
};

export function setLocState(value: any) {
    locState = value;
    if(value.data) {
        loc = {
            ...loc,
            ...locState.data(),
        }
    }
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

let locId = new UUID("aed4c6e4-979e-48ad-be6e-4bd39fb94762");

export function setLocId(id: UUID) {
    locId = id;
}

let loc: Partial<LocData> | null = {
    id: locId,
    locType: "Identity",
    requesterAccountId: ValidAccountId.polkadot("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    userIdentity: {
        firstName: "John",
        lastName: "Doe",
        email: "john.doe@logion.network",
        phoneNumber: "+1234",
    }
};

export function setLocRequest(request: Partial<LocData> | null) {
    loc = request;
}

export let refresh = jest.fn();

export let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

export let deleteMetadata = jest.fn();

export let deleteLink = jest.fn();

let collectionItems: CollectionItem[] = [];

export function setCollectionItems(items: CollectionItem[]) {
    collectionItems = items;
}

async function mutateLocState(mutator: (state: any) => Promise<any>): Promise<void> {
    await mutator(locState);
}

export function useLocContext() {
    return {
        linkLoc: {

        },
        locId,
        loc,
        locItems,
        refresh,
        requestSof,
        deleteMetadata,
        deleteLink,
        checkResult: { result: "NONE" },
        locState,
        collectionItems,
        mutateLocState,
    };
}
