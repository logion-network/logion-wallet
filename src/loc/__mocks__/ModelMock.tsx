export let rejectRequest = jest.fn();

export let acceptRequest = jest.fn();

export function setAcceptRequest(mockFn: any) {
    acceptRequest = mockFn;
}

export let setAssetDescription = jest.fn();

export let rejectProtectionRequest = jest.fn();

export function setRejectProtectionRequest(mockFn: any) {
    rejectProtectionRequest = mockFn;
}

export let acceptProtectionRequest = jest.fn();

export function setAcceptProtectionRequest(mockFn: any) {
    acceptProtectionRequest = mockFn;
}

export function decision(legalOfficerAddress: string, decisions: any[]) {
    return decisions[0];
}

export let fetchRecoveryInfo = jest.fn();

export function setFetchRecoveryInfo(mockFn: any) {
    fetchRecoveryInfo = mockFn;
}

export let acceptLocRequest = jest.fn();

export function setAcceptLocRequest(mockFn: any) {
    acceptLocRequest = mockFn;
}

export let rejectLocRequest = jest.fn();

export function setRejectLocRequest(mockFn: any) {
    rejectLocRequest = mockFn;
}

export let addLink = jest.fn();
