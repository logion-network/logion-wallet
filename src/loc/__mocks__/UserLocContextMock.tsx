import { UUID } from "@logion/node-api/dist/UUID";
import { LocData } from "@logion/client/dist/Loc";

import { LocRequest } from "../../common/types/ModelTypes";
import { LocItem } from "../types";

let locState: any;

export function setLocState(value: any) {
    locState = value;
}

export let refresh = jest.fn();

let requestSof = jest.fn();

export function setRequestSof(fn: jest.Mock) {
    requestSof = fn;
}

let requestSofOnCollection = jest.fn();

export function setRequestSofOnCollection(fn: jest.Mock) {
    requestSofOnCollection = fn;
}

let locId: UUID;

export function setLocId(id: UUID) {
    locId = id;
}

let loc: LocData | null = null;

export function setLoc(request: LocData | null) {
    loc = request;
}

let locItems: LocItem[] = [];

export function setLocItems(items: LocItem[]) {
    locItems = items;
}

export let deleteMetadata = jest.fn();

export let deleteFile = jest.fn();

export let addFile = jest.fn().mockResolvedValue(undefined);

export let addMetadata = jest.fn().mockResolvedValue(undefined);

export function useUserLocContext() {

    return {
        refresh,
        locState,
        requestSof,
        requestSofOnCollection,
        locId,
        loc,
        deleteMetadata,
        deleteFile,
        locItems,
        addFile,
        addMetadata,
    }
}
