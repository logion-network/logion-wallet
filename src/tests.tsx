import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };

export async function clickByName(name: string | RegExp | ((accessibleName: string, element: Element) => boolean)) {
    let button: HTMLElement;
    await waitFor(() => button = screen.getByRole("button", { name }));
    await userEvent.click(button!);
}

export async function typeByLabel(label: string, value: string) {
    let input: HTMLElement;
    await waitFor(() => input = screen.getByLabelText(label));
    await userEvent.type(input!, value);
}
