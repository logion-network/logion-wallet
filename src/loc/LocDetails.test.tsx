import { shallowRender } from "../tests";
import { setParams } from "../__mocks__/ReactRouterMock";
import LocDetails, { UserLocDetails } from "./LocDetails";

jest.mock("react-router");

describe("LocDetails", () => {

    it("renders for LO", () => {
        setParams({"locId": "aed4c6e4-979e-48ad-be6e-4bd39fb94762"})
        const result = shallowRender(<LocDetails
            backPath="/"
            detailsPath={() => "/details"}
        />);
        expect(result).toMatchSnapshot();
    })
})

describe("LocDetails", () => {

    it("renders for user as Requester", () => {
        setParams({"locId": "aed4c6e4-979e-48ad-be6e-4bd39fb94762"})
        const result = shallowRender(<UserLocDetails
            backPath="/"
            detailsPath={() => "/details"}
            contributionMode='Requester'
        />);
        expect(result).toMatchSnapshot();
    })

    it("renders for user as verified issuer", () => {
        setParams({"locId": "aed4c6e4-979e-48ad-be6e-4bd39fb94762"})
        const result = shallowRender(<UserLocDetails
            backPath="/"
            detailsPath={() => "/details"}
            contributionMode='Issuer'
        />);
        expect(result).toMatchSnapshot();
    })
})
