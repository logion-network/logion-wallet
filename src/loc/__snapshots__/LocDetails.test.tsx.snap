// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocDetails renders for LO 1`] = `
<LegalOfficerLocContextProvider
  backPath="/"
  detailsPath={[Function]}
  locId={
    UUID {
      "bytes": Uint8Array [
        174,
        212,
        198,
        228,
        151,
        158,
        72,
        173,
        190,
        110,
        75,
        211,
        159,
        185,
        71,
        98,
      ],
    }
  }
>
  <ContextualizedLocDetails />
</LegalOfficerLocContextProvider>
`;

exports[`LocDetails renders for user as Requester 1`] = `
<UserLocContextProvider
  backPath="/"
  detailsPath={[Function]}
  locId={
    UUID {
      "bytes": Uint8Array [
        174,
        212,
        198,
        228,
        151,
        158,
        72,
        173,
        190,
        110,
        75,
        211,
        159,
        185,
        71,
        98,
      ],
    }
  }
>
  <UserContextualizedLocDetails
    contributionMode="Requester"
  />
</UserLocContextProvider>
`;

exports[`LocDetails renders for user as verified issuer 1`] = `
<UserLocContextProvider
  backPath="/"
  detailsPath={[Function]}
  locId={
    UUID {
      "bytes": Uint8Array [
        174,
        212,
        198,
        228,
        151,
        158,
        72,
        173,
        190,
        110,
        75,
        211,
        159,
        185,
        71,
        98,
      ],
    }
  }
>
  <UserContextualizedLocDetails
    contributionMode="Issuer"
  />
</UserLocContextProvider>
`;
